Neos.Utility.Files/Classes/Files.php

```diff
@@ -432,16 +431,27 @@ public static function unlink($pathAndFilename)
      * @param string $thousandsSeparator thousands separator of the resulting string
      * @return string the size string, e.g. "1,024 MB"
      */
-    public static function bytesToSizeString($bytes, $decimals = 0, $decimalSeparator = '.', $thousandsSeparator = ',')
+    public static function bytesToSizeString($bytes, int $decimals = null, string $decimalSeparator = null, string $thousandsSeparator = null): string
     {
-        if (!is_integer($bytes) && !is_float($bytes)) {
+        if (!is_int($bytes) && !is_float($bytes)) {
             if (is_numeric($bytes)) {
                 $bytes = (float)$bytes;
             } else {
                 $bytes = 0;
             }
         }
 
+        // FIXME: The defaults should again be in the function declaration and nulls not allowed. This is a breaking change though.
+        if ($decimals === null) {
+            $decimals = 0;
+        }
+        if ($decimalSeparator === null) {
+            $decimalSeparator = '.';
+        }
+        if ($thousandsSeparator === null) {
+            $thousandsSeparator = ',';
+        }
+
```

Comment:
OK, thanks. I wasn't aware of the fact that adding a type hint breaks compatibility with `null` :-/
But instead of that work around, personally I'd use:
```php
public static function bytesToSizeString($bytes, $decimals = 0, $decimalSeparator = '.', $thousandsSeparator = ','): string
```
We can't use the type hint for `$bytes` and no `void` return types yet anyways, so no problem in omitting some